src/algebra/algebra/unitization.lean

```diff
@@ -0,0 +1,498 @@
+instance [comm_semiring R] [comm_semiring A] [module R A] [is_scalar_tower R A A]
+  [smul_comm_class R A A] : comm_semiring (unitization R A) :=
+{ ..unitization.comm_monoid,
+  ..unitization.non_assoc_semiring }
+
+variables (R A)
+
+/-- The canonical inclusion of rings `R →+* unitization R A`. -/
+@[simps apply]
+def inl_ring_hom [semiring R] [non_unital_semiring A] [module R A] : R →+* unitization R A :=
+{ to_fun := inl,
+  map_one' := inl_one A,
+  map_mul' := inl_mul A,
+  map_zero' := inl_zero A,
+  map_add' := inl_add A }
+
+end mul
+
+section algebra
+variables (R A : Type*)
+variables [comm_semiring R] [non_unital_semiring A]
+variables [module R A] [is_scalar_tower R A A] [smul_comm_class R A A]
+
+instance algebra : algebra R (unitization R A) :=
+{ commutes' := λ r x,
+  begin
+    induction x using unitization.ind,
+    simp only [ring_hom.to_fun_eq_coe, inl_ring_hom_apply, mul_add, add_mul, inl_mul_inl],
+    rw [inl_mul_inr, inr_mul_inl, mul_comm]
+  end,
+  smul_def' := λ r x,
+  begin
+    induction x using unitization.ind,
+    simp only [mul_add, inl_mul_inr, smul_add, ring_hom.to_fun_eq_coe, inl_ring_hom_apply,
+      inr_smul],
+    rw [inl_mul_inl, ←smul_eq_mul, inl_smul],
+  end,
+  ..(unitization.inl_ring_hom R A) }
+
+lemma algebra_map_eq_inl : ⇑(algebra_map R (unitization R A)) = inl := rfl
+lemma algebra_map_eq_inl_hom : algebra_map R (unitization R A) = inl_ring_hom R A := rfl
+
+/-- The canonical `R`-algebra projection `unitization R A → R`. -/
+@[simps]
+def fst_hom : unitization R A →ₐ[R] R :=
+{ to_fun := fst,
+  map_one' := fst_one,
+  map_mul' := fst_mul,
+  map_zero' := fst_zero,
+  map_add' := fst_add,
+  commutes' := fst_inl A }
+
+end algebra
+
+section coe
+variables {R A : Type*}
+
+instance [has_zero R] : has_coe_t A (unitization R A) := { coe := inr }
+
+@[simp]
+lemma coe_apply [has_zero R] (a : A) : (a : unitization R A) = inr a := rfl
+
+lemma coe_injective [has_zero R] : function.injective (coe : A → unitization R A) :=
+inr_injective
+
+/-- The coercion from a non-unital `R`-algebra `A` to its unitization `algebra.unitization R A`
+realized as a non-unital algebra homomorphism. -/
+@[simps]
+def coe_non_unital_alg_hom [comm_semiring R] [non_unital_semiring A] [module R A] :
+  non_unital_alg_hom R A (unitization R A) :=
+{ to_fun := coe,
+  map_smul' := by simp,
+  map_zero' := by simp,
+  map_add' := by simp,
+  map_mul' := by simp }
+
+end coe
+
+end unitization
+
+section alg_hom
+
+open unitization
+
+variables
+{R A : Type*}
+[comm_semiring R] [non_unital_semiring A]
+[module R A] [smul_comm_class R A A] [is_scalar_tower R A A]
+{B : Type*} [ring B] [algebra R B]
+
+/-- The extension to `unitization R A →ₐ[R] B` of a non-unital algebra homomorphism `φ : A → B`
+from a non-unital `R`-algebra `A` into a unital `R`-algebra `B`. -/
+@[simps]
+def non_unital_alg_hom.to_alg_hom (φ : non_unital_alg_hom R A B) : (unitization R A) →ₐ[R] B :=
```

Comment:
I would be inclined to call this `unitization.lift` to match `with_one.lift`.

Reply:
Can we at least call it `non_unital_alg_hom.lift` so we have dot notation?

Reply:
The phrasing I'd expect is `non_unital_alg_hom R A B ≃ (unitization R A →ₐ[R] B)`, assuming that's true.

I agree dot notation would be nice, but I think for now its better to be consistent with other parts of the library. If we're going to change them for dot notation, we should try to do them all at once.

Reply:
done